decentraland/unity-renderer
# Chore: reduce Graphics.Raycast overhead

## What does this PR change?
Closes #4045 

As described in the ticket above👆 `EventSystem.current.RaycastAll()` call was expensive. To reduce it I did the following:
- Disabled `GraphicRaycaster` component for UI elements with `ShowHideAnimator`, when this element is hidden (mainly in this [commit](https://github.com/decentraland/unity-renderer/pull/4118/commits/fc402ad3beaaa27f1742b53758a3caaad85eecc0#diff-f8718f2ede0413a4a3d5720c51917d03e1098319e632d972dbacb7f7ac242de7))
- Removed second call of `RaycastAll` from our side by re-using the data from EventSystem (presented in this [commit](https://github.com/decentraland/unity-renderer/pull/3891/commits/5d11b56e45b4e5072fdaa5993e9967f734d21fd2)). To do that, I needed 
    1. to create a child class of `StandaloneInputModule` to expose this data and added it to our prefab of event system (in this [commit](https://github.com/decentraland/unity-renderer/pull/4118/commits/ff867ddcb67945f52fdb7b22a4c699a83be79a4c)). 
    2. replace old logic of  the `CanRaycastWhileUnlocked` method inside `PointerEventsController.cs` with the new one that re-uses exposed EventSystem data.
- Disabled `raycastTarget` setting for some UI elements in common UI prefabs

#### Additionally:
- Replaced destroying of `Animator `component in `Awake `of `ShowHideAnimator `by the respective validation test (`PrefabsValidationTests.cs`->`ValidateShowHideAnimators`). And removed such Animator components (and respective assets) to make test pass.
- Several refactorings of mentioned classes to reduce cyclomatic complexity and improve readability

## How to test the changes?

1. Go to: https://play.decentraland.zone/?renderer-branch={branch_name}&{desired_url_params}
2. verify that all UI interactions work as previously (buttons, sliders, scroll bars)
3. verify that passports cannot be open through UI.

[Devs Only]:
1. Open this branch in Editor
2. Open Profiler tab with DeepProfilier enabled
3. Run it at 100,100 parcel
4. Check in profiler Hierarchy that 
    - `RaycastAll` call appears only once 
    - it does not take more than 3 ms 
    - it has less GraphicRaycaster calls

#### **Before**
![image](https://user-images.githubusercontent.com/35366872/214965399-a942fdb2-af62-43fb-bada-16fe77a38529.png)

#### **After**
![image](https://user-images.githubusercontent.com/35366872/214964191-c298c842-fe01-4caf-ad32-e20322dbc2e6.png)

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
